src/main/java/org/broadinstitute/hellbender/tools/spark/pathseq/PathSeqFilterSpark.java

```diff
@@ -0,0 +1,192 @@
+
+        //Filter using base quality scores
+        ReadFilter bqFilter = new BaseQualityReadFilter(QUAL_PHRED_THRESH,QUAL_MAX_FRAC);
+        final JavaRDD<GATKRead> readsBQFiltered = readsClipped.filter(read -> bqFilter.test(read));
+
+        //Filter reads with ambiguous bases
+        ReadFilter ambigFilter = new AmbiguousBaseReadFilter(FRAC_N_THRESHOLD);
+        final JavaRDD<GATKRead> readsAmbigFiltered = readsBQFiltered.filter(read -> ambigFilter.test(read));
+
+        //Filter reads with less than MIN_READ_LENGTH bases
+        ReadFilter readLengthFilter = new ReadLengthReadFilter(MIN_READ_LENGTH,MAX_READ_LENGTH);
+        final JavaRDD<GATKRead> readsLengthFiltered = readsAmbigFiltered.filter(read -> readLengthFilter.test(read));
+
+        //Load Kmer hopscotch set and filter reads containing > 0 matching kmer
+        final PipelineOptions options = getAuthenticatedGCSOptions();
+        Input input = new Input(BucketUtils.openFile(KMER_LIB_PATH, options));
+        Kryo kryo=new Kryo();
+        HopscotchSet<SVKmerSmall> kmerLibSet = (HopscotchSet<SVKmerSmall>)kryo.readClassAndObject(input);
+        Broadcast<HopscotchSet<SVKmerSmall>> kmerLibBroadcast = ctx.broadcast(kmerLibSet);
+        ContainsKmerReadFilter readKmerFilter = new ContainsKmerReadFilter(kmerLibBroadcast,KMER_SIZE);
+        final JavaRDD<GATKRead> readsKmerFiltered = readsLengthFiltered.filter(read -> !readKmerFilter.test(read));
+
+        //TODO: bwa filtering against user-specified reference
+
+        //"Clean" reads by resetting all flags and setting pairedness flags according to the remaining read set
+        final JavaRDD<GATKRead> readsCleaned = readsKmerFiltered
+                .groupBy(read -> read.getName()) //group pairs using read name into set of Iterable<GATKRead>'s
+                .values()   //Get Iterables
+                .map(p -> {     //Determine if Iterable represents 1 or 2 reads and set flags accordingly
+                    int size = 0;
+                    Iterator<GATKRead> iter = p.iterator();
+                    while (iter.hasNext()) {
+                        size++;
+                        iter.next();
+                    }
+                    boolean notPair = (size==1);
+                    ArrayList<GATKRead> newPair = new ArrayList<>(size);
+                    iter = p.iterator();
+                    while (iter.hasNext()) {
+                        newPair.add(CleanRead(iter.next(),notPair));
+                    }
+                    return newPair;
+                })
+                .flatMap(p -> p);   //Flatten back from Iterables to GAKTReads
+
+        writeReads(ctx, output, readsCleaned);
+    }
+
+    private boolean BaseQualityFilterFn(GATKRead read) {
+        return false;
+    }
+
+    private GATKRead CleanRead(GATKRead read, boolean notPaired) {
+
+        GATKRead new_read = new SAMRecordToGATKReadAdapter(new SAMRecord(header));
```

Comment:
I'm pretty sure you can pass `null` for a header as long as you avoid methods that try to turn reference names into reference indices or vice versa (and since you're producing unmapped reads, you should be good to go).  If I'm right about that, then this method can be made static and you won't capture the `this` pointer in the lambda that calls CleanRead.  This header/headerless, reference name/index morass is one of my least favorite parts of GATK.